src/features/ProfilePage/index.tsx

```diff
@@ -0,0 +1,76 @@
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+
+import { selectUserRole } from '../Authentication/userSlice';
+import { useAppSelector } from '@/store';
+
+import AccountInfo from './components/AccountInfo';
+import {
+  CONTENT_WIDTH,
+  OUTER_VERTICAL_MARGIN,
+  palette,
+} from '@/components/variables';
+import PageWithTransition from '@/components/PageWithTransition';
+import PublicInfo from './components/PublicInfo';
+import Text from '@/components/Text';
+
+import type { UserRole } from '../Authentication/authenticationApi';
+
+const ProfilePage = () => {
+  const { t } = useTranslation('profile');
+  const storedUserRole = useAppSelector(selectUserRole);
+
+  const convertUserRole = () => {
+    if (storedUserRole === 'admin') return 'admin';
+    if (storedUserRole === 'mentor') return 'mentor';
+    return 'mentee';
+  };
+  const userRole: UserRole = convertUserRole();
```

Comment:
Instead of this piece of code, I would create another selector for this

```typescript
export const selectApiRole = createSelector(
  selectUserState,
  ({ userRole }) =>
    userRole ?? 'mentee'
);
```
This seems a bit weird workaround in the UI layer

btw, i dont know about the name. Maybe the selectUserRole should be renamed to `selectAppRole`

Reply:
Yeah, you're right, that is some weird sh*t...  😅

Reply:
I went with selectUserRole for that one (roles from the API, defaulting to _mentee_) and selectAppRole for the one that has the _freshMentee_ option.